src/System.Text.Json/src/System/Text/Json/Serialization/CachedUtf8JsonWriter.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+
+namespace System.Text.Json.Serialization
+{
+    internal sealed class CachedUtf8JsonWriter
+    {
+        [ThreadStatic]
+        private static CachedUtf8JsonWriter s_cachedInstance;
+
+        private readonly Utf8JsonWriter _writer;
+
+#if DEBUG
+        private bool _inUse;
+#endif
+
+        private CachedUtf8JsonWriter(IBufferWriter<byte> stream, JsonWriterOptions writerOptions)
```

Comment:
nit: I would rename `stream` to `output` for the parameter name for `IBufferWriter<byte>`.

Here/elsewhere.